server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -251,6 +263,10 @@ def _validate_and_initialize(self):
         self._validate_label_data()
         self._create_schema()
 
+        # heuristic
+        if (self.data.shape[0] * self.data.shape[1]) > (100 * 10 ** 6):
```

Comment:
```suggestion
        n_values = self.data.shape[0] * self.data.shape[1]
        if (n_values > 1e8 and self.config['backed']==True) or (n_values > 5e8):
```

Reply:
Still a hacky heuristic. But, I found that tmuris (n_values ~= 3e8) was fine in not-backed mode and slow in backed mode. 1.3M dataset (n_values ~= 1e9) was fatally slow in both cases.

Reply:
fine with me - I'll make this change.